app/lib/search/index_simple.dart

```diff
@@ -141,6 +146,43 @@ class SimplePackageIndex implements PackageIndex {
     _isReady = true;
     _lastUpdated = new DateTime.now().toUtc();
   }
+
+  void _clearScores() {
+    _healthScores = null;
+    _popularityScores = null;
+  }
+
+  void _initScoresIfNeeded() {
+    if (_healthScores != null) return;
+    _healthScores = <String, double>{};
+    _popularityScores = <String, double>{};
+
+    final healthScorer =
+        new scoring.Summary(_documents.values.map((pd) => pd.health ?? 0.0));
+    final popularityScorer = new scoring.Summary(
+        _documents.values.map((pd) => pd.popularity ?? 0.0));
+    for (PackageDocument doc in _documents.values) {
+      if (doc.health != null) {
+        _healthScores[doc.url] = healthScorer.bezierScore(doc.health) * 100.0;
+      }
+      if (doc.popularity != null) {
+        _popularityScores[doc.url] =
+            popularityScorer.bezierScore(doc.popularity) * 100.0;
```

Comment:
Here I would assume that our bigquery batch job gives us already normalized popularity metric, something like `#of-package-X-downloads / #of-total-downloads` (or something more complicated), scaled to an appropriate interval, like `[0, 1]`.